crab/cli.py

```diff
@@ -78,12 +79,20 @@ def main(command=None):
         port = env.setdefault("PORT", get_free_port())
         command = [item.replace("$PORT", port) for item in command]
 
-    if shutil.which(command[0], path=env["PATH"]) is None:
-        print('Could not find "{}" in your procfile or $PATH.'.format(command[0]))
-        exit(1)
+    # Is it a shell function?
+    if (
+        subprocess.run(
+            ["declare", "-f", command[0]], shell=True, stdout=subprocess.DEVNULL
+        ).returncode
+        == 0
+    ):
+        return os.execvpe(env["SHELL"], [env["SHELL"], "-ci"] + command, env)
+
+    if shutil.which(command[0], path=env["PATH"]) is not None:
+        return os.execvpe(command[0], command, env)
 
-    # off we go
-    os.execvpe(command[0], command, env)
+    print('Could not find "{}" in your procfile or $PATH.'.format(command[0]))
```

Comment:
Worth mentioning here about it also not being a shell function?

Although that would be quite a long message. Shells just do "command not found".